source/developers/development_environment.markdown

```diff
@@ -11,20 +11,55 @@ footer: true
 
 You'll need to set up a development environment if you want to develop a new feature or component for Home Assistant. Read on to learn how to set up.
 
-* Visit the [Home Assistant repository](https://github.com/home-assistant/home-assistant) and click **Fork**.
-
-* Consider setting up a virtual environment using [`venv`](https://docs.python.org/3.4/library/venv.html) before running the setup script.
+### {% linkable_title Setup Local Repository %}
 
+Visit the [Home Assistant repository](https://github.com/home-assistant/home-assistant) and click **Fork**.
+Once forked, setup your local copy of the source using the commands:
 ```bash
 $ git clone https://github.com/YOUR_GIT_USERNAME/home-assistant.git
 $ cd home-assistant
 $ git remote add upstream https://github.com/home-assistant/home-assistant.git
+```
+
+### {% linkable_title Prepare Your Environment %}
+
+#### {% linkable_title Core dependencies %} 
+
+Install the core dependencies.
+```bash
+$ sudo apt-get install python3-pip python3-dev libssl-dev
+```
+
+#### {% linkable_title Frontend dependencies (optional) %} 
```

Comment:
Doesn't this belong on the dedicated frontend dev page?
https://home-assistant.io/developers/frontend/